File: Brackets (Easy Version) - 339641C/sol.py
s = input()
if s == "":
    print("YES")
else:
    sum = 0
    open_left = 0
    open_right = 0
    for i in s:
        if i == '(':
            sum += 1
            open_right += 1
        elif i == ')' and sum > 0:
            sum -= 1
        if i == ')':
            open_left += 1
    if sum == 0 and open_left == open_right:
        print("YES")
    else:
        print("NO")

File: Brackets (Easy Version) - 339641C/sol.cpp
#include <iostream>
#include <bits/stdc++.h>
#include<numeric>
#include <algorithm>
#define ll long long int
using namespace std;
void Fast()
{
    cin.tie(0);
    cin.sync_with_stdio(0);
}

int main()
{
    Fast(); //File_Input();
    string s; cin >> s;
    if(s.empty())
        cout << "YES\n";
    else
    {
        int counter = 0, openLeft = 0, openRight = 0;
        for (auto &i: s)
        {
            if(i == '(')
            {
                counter ++;
                openRight ++;
            }
            else if(i == ')' && counter > 0)
                counter --;
            if(i == ')')
                openLeft ++;
        }
        if(counter == 0 and openLeft == openRight)
            cout << "YES";
        else
            cout << "NO";
    }
    return 0;
}
